python/az/aro/azext_aro/tests/latest/unit/test_validators.py

```diff
@@ -0,0 +1,419 @@
+def test_validate_client_secret(test_description, isCreate, namespace, expected_exception):
+    validate_client_secret_fn = validate_client_secret(isCreate)
+    if expected_exception is None:
+        validate_client_secret_fn(namespace)
+    else:
+        with pytest.raises(expected_exception):
+            validate_client_secret_fn(namespace)
+
+
+test_validate_cluster_resource_group_data = [
+    ("should not raise any exception when namespace.cluster_resource_group is None", None, None, Mock(cluster_resource_group=None), None),
+    ("should raise InvalidArgumentValueError exception when namespace.cluster_resource_group is not None and resource group exists in the client returned by get_mgmt_service_client", Mock(**{"resource_groups.check_existence.return_value": True}), Mock(cli_ctx=None), Mock(cluster_resource_group="some_resource_group"), InvalidArgumentValueError),
+    ("should not raise any exception when namespace.cluster_resource_group is not None and resource group does not exists in the client returned by get_mgmt_service_client", Mock(**{"resource_groups.check_existence.return_value": False}), Mock(cli_ctx=None), Mock(cluster_resource_group="some_resource_group"), None)
+]
+
+
+@pytest.mark.parametrize("test_description, client_mock, cmd_mock, namespace, expected_exception", test_validate_cluster_resource_group_data, ids=[i[0] for i in test_validate_cluster_resource_group_data])
+@patch('azext_aro._validators.get_mgmt_service_client')
+def test_validate_cluster_resource_group(get_mgmt_service_client_mock, test_description, client_mock, cmd_mock, namespace, expected_exception):
+    get_mgmt_service_client_mock.return_value = client_mock
+    if expected_exception is None:
+        validate_cluster_resource_group(cmd_mock, namespace)
+    else:
+        with pytest.raises(expected_exception):
+            validate_cluster_resource_group(cmd_mock, namespace)
+
+
+test_validate_disk_encryption_set_data = [
+    ("should not raise any exception when namespace.disk_encryption_set is None", None, Mock(disk_encryption_set=None), None, None, None, None),
+    ("should raise InvalidArgumentValueError exception when namespace.disk_encryption_set is not None and is_valid_resource_id(namespace.disk_encryption_set) returns False", None, Mock(disk_encryption_set="something different than None"), False, None, InvalidArgumentValueError, None),
+    ("should not raise any exception when compute_client.disk_encryption_sets.get() not raises CludError exception", Mock(cli_ctx=None), Mock(disk_encryption_set="something different than None"), True, Mock(), None, {"resource_group": None, "name": None})
+]
+
+
+@pytest.mark.parametrize("test_description, cmd_mock, namespace, is_valid_resource_id_return_value, compute_client_mock, expected_exception, parse_resource_id_mock_return_value", test_validate_disk_encryption_set_data, ids=[i[0] for i in test_validate_disk_encryption_set_data])
```

Comment:
By this point we have 7 ordered parameters to our test suite and I'm having difficulty parsing out or remembering what each ordered item is. This makes me think this "unit" is actually doing too much. Can we simplify the code so it is testable?

Reply:
I would say this should be a different PR. For now, having those 74 test cases is a good starting point. 😃